### ISSUE TITLE
Tye build project c# with exiting Dockerfile

### ISSUE THREAD

**Issue by thinhnotes** [dotnet/tye#998]
### Describe the bug
In current I had a issues when I used the tye with run project c# with DockerFile

### To Reproduce
example in your source code in samples/azure-functions/voting

I edit file tye.yaml in bellow:

```
name: VotingSample
services:
- name: worker
  project: worker-function/worker-function.csproj
```

run tye build -v debug

and it return the error:
![image](https://user-images.githubusercontent.com/4660531/113405544-3535cc00-93d4-11eb-8e35-9a6509dba5b9.png)

In current I guest it can mapping the context of build folder to docker build and not file the profile file build
I had try the property a dockerFileContext  but it not working on this case

Could you please to help me to resolve it

**Comment by jkotalik** [comment]
Yeah so tye will automatically grab the DockerFile from the folder if one is present. I think we need to add an option to ignore the docker file present; I've hit this issue as well. 

As a workaround for now, you may need to rename your dockerfile present in the directory.

**Comment by thinhnotes** [comment]
thanks for support, I will be try the workaround solution and wait the new info when the option to ignore the docker file is release, thanks

**Comment by philliphoff** [comment]
I think the question is whether the user really *wants* Tye to use the existing `Dockerfile` by default, if present.  In other words, is the user only telling Tye to ignore the existing `Dockerfile` because Tye doesn't do the right thing with it and not for any other reason.  Would the user prefer to make the choice "opt in", say by using the existing `dockerFile`/`dockerContext` configuration properties (which seem to be ignored for `tye build`), and otherwise have Tye generate its own `Dockerfile` for build and deployment purposes?  If the user's gone to the trouble of creating a Dockerfile, presumably the user knows better than Tye how images should be built.  Therefore, I'm inclined to side with Tye respecting the existing the `Dockerfile` by default.

Then the question is what "kind" of `Dockerfile` should Tye expect.  Currently `tye build` assumes a `Dockerfile` that expects to have a locally-published service copied to it.  However, current `Dockerfile` practice (for .NET) is to use multiple stages to build, publish, and then copy the published output to the "final" image.  There is, in fact, existing (but latent) code in Tye to generate/use a "multiphase" *[sic]* `Dockerfile` that more closely matches current practice.  It may be enough to simply use that flag, either properly inferred from the `Dockerfile` or exposed in the `tye build` command and/or service configuration.  This is the approach I'd be inclined to use and I'd even be inclined to outright swap any default to multi-stage rather than "single stage".  (Doing *just* that may be only thing necessary to resolve the issue for the majority of users.)

**Comment by christiannagel** [comment]
I just had a scenario where it would be great to ignore the existing `Dockerfile`.

With a solution I created, a `Dockerfile` exists for deploying to Azure Container apps. To deploy it to AKS with Tye it would be great to specify to ignore the existing `Dockerfile`.
Using the existing `Dockerfile` failed because the *docker context* can't be specified.
